assets/search-form.js

```diff
@@ -0,0 +1,23 @@
+class SearchForm extends HTMLElement {
+  constructor() {
+    super();
+    this.input = this.querySelector('input[type="search"]');
+
+    this.setupEventListeners();
+  }
+  setupEventListeners() {
+    const form = this.querySelector('form.search');
+    form.addEventListener('reset', this.onFormReset.bind(this));
+  }
+
+  onFormReset(event) {
+    // Prevent default so the form reset doesn't set the value gotten from the url on page load
+    event.preventDefault();
+    // Don't reset if the user has selected an element on the predictive search dropdown
+    if (!document.querySelector('a[aria-selected="true"')) {
+      this.input.value = '';
+    }
+  }
+}
+
+customElements.define('search-form', SearchForm);
```

Comment:
The `extend` is a very nice idea, I'll refactor that.

What do you think about the scope of each file, any concern there? I made them assuming the only moment that there is more than one search input is when the main search is visible, that's why most of the logic is on that file. But if we don't assume that, then most of the logic will move to `search-form.js`. That means loading a bigger file with the header (with code that is not really needed) but also we will have a more resilient code in case that assumption changes in the future. What do yo think?

Reply:
That's a good point. Maybe it is better for now to keep it as you have it 🤔 Only load that script when actually needed. Since it's on the search page anyway it shouldn't be too impactful. 👍